client/contracts/bindings/consensus/consensus.go

```diff
@@ -239,84 +217,76 @@ func (consensus *consensus) Deposits(addr common.Address) ([]*types.Deposit, err
-	return consensus.manager.GetMinimumDeposit(&bind.CallOpts{})
+func (css *Consensus) MinimumDeposit() (*big.Int, error) {
+	return css.manager.GetMinimumDeposit(&bind.CallOpts{})
 }
 
-func (consensus *consensus) GetValidatorCount() (*big.Int, error) {
-	return consensus.manager.GetValidatorCount(&bind.CallOpts{})
+func (css *Consensus) GetValidatorCount() (*big.Int, error) {
+	return css.manager.GetValidatorCount(&bind.CallOpts{})
 }
 
-func (consensus *consensus) MaxValidators() (*big.Int, error) {
-	return consensus.manager.MaxNumValidators(&bind.CallOpts{})
+func (css *Consensus) MaxValidators() (*big.Int, error) {
+	return css.manager.MaxNumValidators(&bind.CallOpts{})
 }
 
-func (consensus *consensus) Token() token.Token {
-	return consensus.mtoken
+func (css *Consensus) Token() token.Token {
+	return css.mtoken
 }
 
 //Minter interface implementation
 
-func (consensus *consensus) MintInit() error {
+func (css *Consensus) MintInit() error {
 	var err error
-	consensus.initMint.Do(func() {
-		if consensus.multiSigWallet == nil {
-			addr, ok := mapMiningTokenToAddr[consensus.chainID.Uint64()]
-			if !ok {
-				err = bindings.ErrNoAddress
-				return
-			}
-			consensus.mtokenAddr = addr
-
-			addr, ok = mapMultiSigWalletToAddr[consensus.chainID.Uint64()]
-			if !ok {
-				err = bindings.ErrNoAddress
-				return
-			}
+	css.initMint.Do(func() {
+		if css.multiSigWallet == nil {
+			addr := common.HexToAddress(bindings.MultiSigWalletAddr)
 
 			var multisig *ownership.MultiSigWallet
-			multisig, err = ownership.NewMultiSigWallet(addr, consensus.contractBackend)
+			multisig, err = ownership.NewMultiSigWallet(addr, css.contractBackend)
 			if err != nil {
 				return
 			}
 
-			consensus.multiSigWallet = multisig
+			css.multiSigWallet = multisig
 		}
 
-		if consensus.oracle == nil {
-			addr, ok := mapOracleToAddr[consensus.chainID.Uint64()]
-			if !ok {
-				err = bindings.ErrNoAddress
+		if css.oracle == nil {
+			addr, errKns := kns.GetAddressFromDomain(
+				params.KNSDomains[params.OracleMgrDomain].FullDomain(),
+				css.contractBackend,
+			)
+			if err != nil {
+				err = errKns
 				return
 			}
 
 			var oracleMgr *oracle.OracleMgr
-			oracleMgr, err = oracle.NewOracleMgr(addr, consensus.contractBackend)
+			oracleMgr, err = oracle.NewOracleMgr(addr, css.contractBackend)
 			if err != nil {
 				return
 			}
 
-			consensus.oracle = oracleMgr
+			css.oracle = oracleMgr
 		}
 	})
 
 	return err
 }
 
-func (consensus *consensus) MultiSigWalletContract() *ownership.MultiSigWallet {
-	return consensus.multiSigWallet
+func (css *Consensus) MultiSigWalletContract() *ownership.MultiSigWallet {
+	return css.multiSigWallet
 }
 
-func (consensus *consensus) Mint(opts *accounts.TransactOpts, to common.Address, value *big.Int) (common.Hash, error) {
-	if err := consensus.MintInit(); err != nil {
+func (css *Consensus) Mint(opts *accounts.TransactOpts, to common.Address, value *big.Int) (common.Hash, error) {
```

Comment:
This code does not have anything to do with the consensus - should be in a different section.